diff -Naur fte.orig/install fte/install
--- fte.orig/install	2004-04-12 10:39:50.000000000 -0300
+++ fte/install	2007-06-06 10:30:46.000000000 -0300
@@ -12,7 +12,7 @@
 # INSTALLATION CODE (do not touch):
 
 BINDIR=$PREFIX/bin
-LIBDIR=$PREFIX/lib/fte
+LIBDIR=$PREFIX/lib64/fte
 CONFIGDIR=$LIBDIR/config
 
 LOCALCONFIGDIR=$LIBDIR/localconfig
diff -Naur fte.orig/Makefile fte/Makefile
--- fte.orig/Makefile	2004-04-12 10:39:50.000000000 -0300
+++ fte/Makefile	2007-06-06 10:30:39.000000000 -0300
@@ -1,7 +1,7 @@
 PREFIX=/usr/local
 
 BINDIR=$(PREFIX)/bin
-LIBDIR=$(PREFIX)/lib/fte
+LIBDIR=$(PREFIX)/lib64/fte
 CONFIGDIR=$(LIBDIR)/config
 
 .PHONY: all install
