--- fte/src/con_slang.cpp.slang2_compat	2004-04-12 21:39:52.000000000 +0800
+++ fte/src/con_slang.cpp	2011-01-21 23:37:25.000000000 +0800
@@ -74,6 +74,10 @@
 #define DCH_SLANG_ALEFT      147
 #define DCH_SLANG_ARIGHT     148
 
+#define SLSMG_EXTRACT_CHAR_FTE(x) ((x) & 0xFF)
+#define SLSMG_EXTRACT_COLOR_FTE(x) (((x)>>8)&0xFF)
+#define SLSMG_BUILD_CHAR_FTE(ch,color) (((unsigned short)(unsigned char)(ch))|((color)<<8))
+
 static char slang_dchs[] =
 {
     'l',
@@ -215,7 +219,7 @@
 int ConInit(int /*XSize */ , int /*YSize */ )
 {
     unsigned i;
-    unsigned short linebuf[sizeof(slang_dchs)];
+    SLsmg_Char_Type linebuf[sizeof(slang_dchs)];
 
     SLtt_get_terminfo();
 
@@ -248,7 +252,7 @@
     SLsmg_gotorc(0, 0);
     SLsmg_read_raw(linebuf, sizeof(slang_dchs));
     for (i = 0; i < sizeof(slang_dchs); i++)
-	raw_dchs[i] = (linebuf[i]) & 0xff;
+	raw_dchs[i] = SLSMG_EXTRACT_CHAR(linebuf[i]);
 
     SLsmg_set_char_set(0);
 
@@ -363,12 +367,20 @@
 
 static int ConPutBoxRaw(int X, int Y, int W, int H, unsigned short *box)
 {
-    int CurX, CurY;
+    int CurX, CurY, cW;
+    SLsmg_Char_Type t[W], *ct;
 
     ConQueryCursorPos(&CurX, &CurY);
     while (H > 0) {
 	SLsmg_gotorc(Y++, X);
-	SLsmg_write_raw(box, W);
+	ct = t;
+	for (cW = 0; cW < W; ++cW) {
+	    ct->nchars = 1;
+	    ct->wchars[0] = SLSMG_EXTRACT_CHAR_FTE(box[cW]);
+	    ct->color = SLSMG_EXTRACT_COLOR_FTE(box[cW]);
+	    ct++;
+	}
+	SLsmg_write_raw(t, W);
 	box += W;
 	H--;
     }
@@ -382,11 +394,18 @@
 {
     int CurX, CurY, i;
     char ch;
+    SLsmg_Char_Type t[W];
+    PCell cc;
 
     ConQueryCursorPos(&CurX, &CurY);
     while (H > 0) {
 	SLsmg_gotorc(Y++, X);
-	SLsmg_read_raw(Cell, W);
+	SLsmg_read_raw(t, W);
+	cc = Cell;
+	for (i = 0; i < W; i++) {
+	    *cc = SLSMG_BUILD_CHAR_FTE(SLSMG_EXTRACT_CHAR(t[i]),SLSMG_EXTRACT_COLOR(t[i]));
+	    cc++;
+	}
 	for (i = 0; i < W; i++)
 	    if (Cell[i] & 0x8000) {
 		ch = Cell[i] & 0xff;
@@ -404,12 +423,19 @@
 
 static int ConGetBoxRaw(int X, int Y, int W, int H, unsigned short *box)
 {
-    int CurX, CurY;
+    int CurX, CurY, cW;
+    SLsmg_Char_Type t[W];
+    unsigned short *cbox;
 
     ConQueryCursorPos(&CurX, &CurY);
     while (H > 0) {
 	SLsmg_gotorc(Y++, X);
-	SLsmg_read_raw(box, W);
+	SLsmg_read_raw(t, W);
+	cbox = box;
+	for (cW = 0; cW < W; ++cW) {
+		*cbox = SLSMG_BUILD_CHAR_FTE(SLSMG_EXTRACT_CHAR(t[cW]),SLSMG_EXTRACT_COLOR(t[cW]));
+		cbox++;
+	}
 	box += W;
 	H--;
     }
